android/src/androidTestLite/java/bbct/android/lite/provider/NavigateUpTest.java

```diff
@@ -0,0 +1,356 @@
+package bbct.android.lite.provider;
+
+
+import android.support.test.espresso.ViewInteraction;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.test.suitebuilder.annotation.LargeTest;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewParent;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import bbct.android.common.R;
+
+import static android.support.test.InstrumentationRegistry.getInstrumentation;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.Espresso.openActionBarOverflowOrOptionsMenu;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withClassName;
+import static android.support.test.espresso.matcher.ViewMatchers.withContentDescription;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.is;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class NavigateUpTest {
+
+    @Rule
+    public ActivityTestRule<LiteActivity> mActivityTestRule = new ActivityTestRule<>(LiteActivity.class);
+
+    @Test
+    public void navigateUpTest() {
```

Comment:
Break this into multiple smaller tests. There should be a single test for each Fragment.

Reply:
okay

Reply:
This Thread.sleep() waits for 5 minutes while the entire test suite runs in about 20 minutes from what I remember. That adds 25% running time to the entire test suite which is undesirable.